lib/bamboo/mailer.ex

```diff
@@ -254,6 +254,41 @@ defmodule Bamboo.Mailer do
     Formatter.format_email_address(record, %{type: type})
   end
 
+  @doc ~S"""
+  Transforms map of headers to list of tuples.
+
+  Headers with list values are transformed, by default, into a single comma
+  separated string. Passing `:list` as the second argument, however, produces a
+  distinct tuple for every value in the header value list.
+  """
+  def normalize_headers(%{headers: _} = email) do
```

Comment:
Should we be calling this function from `validate_and_normalize` in line [185](https://github.com/thoughtbot/bamboo/pull/472/files#diff-a135ce4fd496f56c99cdb2417fa683d4ff6f89ed8d755e8e9b7fbd674af1d337R185) from which we call`normalize_addresses`? 

As it stands, we've given users the ability to set different types of headers, but they have to explicitly call `normalize_headers/2`? Is there a default we could call? Or does the user need to determine how they send headers?